File: backend/generate-telegram-session.mjs
import { TelegramClient } from "telegram";
import { StringSession } from "telegram/sessions/index.js";
import readline from "readline";

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

const question = (prompt) =>
  new Promise((resolve) => rl.question(prompt, resolve));

async function generateSession() {
  console.log("\n=== Telegram Session Generator ===\n");

  const apiId = await question(
    "Enter your TELEGRAM_API_ID (from my.telegram.org): "
  );
  const apiHash = await question(
    "Enter your TELEGRAM_API_HASH (from my.telegram.org): "
  );
  const phoneNumber = await question(
    "Enter your Telegram phone number (with country code, e.g., +919876543210): "
  );

  try {
    console.log("\nConnecting to Telegram...");

    const client = new TelegramClient(
      new StringSession(""),
      Number(apiId),
      apiHash,
      { connectionRetries: 5 }
    );

    await client.start({
      phoneNumber: async () => phoneNumber,
      password: async () => {
        return await question(
          "Enter your 2FA password (or press Enter if you don't have one): "
        );
      },
      onError: (err) => console.error("Error:", err),
    });

    const session = client.session.save();

    console.log("\n✅ Session generated successfully!\n");
    console.log("=== COPY EVERYTHING BELOW ===\n");
    console.log(session);
    console.log("\n=== PASTE INTO .env AS: ===\n");
    console.log(`TELEGRAM_SESSION=${session}`);
    console.log("\n");

    await client.disconnect();
    rl.close();
  } catch (error) {
    console.error("Error:", error.message);
    rl.close();
  }
}

generateSession();

File: backend/generate-telegram-session.py
#!/usr/bin/env python3
"""
Telegram Session Generator for GramJS
This script generates a session string you can use in your .env file
"""

import sys
import asyncio
from telethon import TelegramClient
from telethon.sessions import StringSession

async def generate_session():
    api_id = input("Enter TELEGRAM_API_ID: ").strip()
    api_hash = input("Enter TELEGRAM_API_HASH: ").strip()
    phone = input("Enter your Telegram phone (+country code): ").strip()
    
    if not api_id or not api_hash or not phone:
        print("❌ Missing credentials!")
        return
    
    try:
        api_id = int(api_id)
    except ValueError:
        print("❌ API_ID must be a number!")
        return
    
    print("\n⏳ Connecting to Telegram...\n")
    
    # Create client with empty session first time
    client = TelegramClient(StringSession(), api_id, api_hash)
    
    try:
        # Connect and request code
        await client.connect()
        
        # Send code request
        result = await client.send_code_request(phone)
        print(f"✅ Code sent to your Telegram app!")
        print(f"   (check your Telegram for the 5-digit code)\n")
        
        # Get code from user
        code = input("Enter the verification CODE (5 digits): ").strip()
        
        if not code:
            print("❌ Code cannot be empty!")
            await client.disconnect()
            return
        
        # Sign in with code
        try:
            await client.sign_in(phone, code)
        except Exception as e:
            # May need 2FA password
            if "password" in str(e).lower():
                password = input("Enter your 2FA password: ").strip()
                await client.sign_in(password=password)
            else:
                raise
        
        # Get the session string
        session_str = client.session.save()
        
        print("\n" + "="*50)
        print("✅ SESSION GENERATED SUCCESSFULLY!\n")
        print("="*50)
        print("\nCOPY THIS ENTIRE STRING:\n")
        print(session_str)
        print("\n" + "="*50)
        print("\nThen paste in your .env file as:\n")
        print(f"TELEGRAM_SESSION={session_str}")
        print("\n" + "="*50)
        
        await client.disconnect()
        
    except Exception as e:
        print(f"❌ Error: {str(e)}")
        try:
            await client.disconnect()
        except:
            pass

if __name__ == "__main__":
    asyncio.run(generate_session())
